File: apps/api/athera_api/schemas/synthesis.py
"""عقود طبقة التركيب | Synthesis contracts (PUBRIVA).

**العقد يحمل الحدَّ مع الدعوى.** فلا حقل هنا يخرج فيه وصفُ فجوةٍ بلا
`sources_considered` و`search_scope` و`known_limitations_ar` بجانبه: واجهةٌ
تعرض الوصف وحده تُنتج جملةً مطلقة من عقدٍ محدود.

**والاسم يُعرض لا المفتاح.** كل مفردةٍ تخرج ومعها اسمها العربي ومعناها حيث
كان للمعنى أثر — فلا يقرأ الباحث `weak_signal` ولا `topic_cluster` ولا
`GapGraphNode`.

**والأنماط تُشتقّ من السجل.** مفردةٌ تُكتب مرّتين تفترق بأول إضافة، فيقبل
العقد ما يرفضه القيد — أو العكس، وهو أسوأ.
"""
from __future__ import annotations

import datetime as dt
import uuid

from pydantic import BaseModel, Field

from ..models.synthesis import (
    CONFLICT_KINDS,
    DECIDABLE_STATUSES,
    EFFECT_DIRECTIONS,
    GAP_SOURCE_ROLES,
    GAP_STRENGTHS,
    GAP_TYPES,
    SIGNIFICANCE_STATES,
    SOURCE_SCOPES,
    SUPPORT_ROLES,
    SYNTHESIS_STATUSES,
    THEME_BASES,
)

_DECISION_PATTERN = "^(" + "|".join(DECIDABLE_STATUSES) + ")$"


class DecisionRequest(BaseModel):
    """حكمُ الباحث على مرشَّح — **و`generated` ليست منها**.

    قبولُها مُدخَلًا يعني «أعِد المرشَّح إلى ما قبل أن ينظر فيه أحد»، وهو
    محوُ مراجعةٍ وقعت. والأربع الباقية تشمل «لا أعرف» (ترحيل 0016): من راجع
    ولم يستطع الحكم **لم يرفض**.
    """

    status: str = Field(pattern=_DECISION_PATTERN)
    # سببٌ اختياري يكتبه الباحث لنفسه ولمن يراجع بعده.
    note_ar: str | None = Field(default=None, max_length=2000)


class VocabularyEntry(BaseModel):
    """مفردةٌ باسمها ومعناها — والمعنى ليس زينة."""

    key: str
    label_ar: str
    label_en: str
    meaning_ar: str | None = None


class EvidenceLink(BaseModel):
    """حلقةٌ في سلسلة الأثر: مرجع ← خلية ← شاهد.

    و`matrix_cell_id` هو ما يجعل «اضغط لترى الشاهد» ممكنًا؛ وغيابه يعني أن
    السند من بياناتٍ وصفية لا من محتوًى مقروء — وتلك حالٌ تُعلَن لا تُخفى.
    """

    source_id: uuid.UUID
    title: str
    role: str = Field(pattern="^(" + "|".join(sorted(
        set(SUPPORT_ROLES) | set(GAP_SOURCE_ROLES))) + ")$")
    basis_field_key: str | None = None
    matrix_cell_id: uuid.UUID | None = None
    evidence_scope: str = Field(pattern="^(" + "|".join(SOURCE_SCOPES) + ")$")
    evidence_quote: str | None = None
    evidence_locator: str | None = None
    cell_state: str | None = None
    cell_value_ar: str | None = None


# ═════════════════════ الموضوعات ═════════════════════

class ThemeView(BaseModel):
    """موضوعٌ مرشَّح — **وأساسه معروضٌ لا مضمر**."""

    id: uuid.UUID
    label_ar: str
    description_ar: str | None = None
    basis: str = Field(pattern="^(" + "|".join(THEME_BASES) + ")$")
    basis_label_ar: str
    # الجملة التي تمنع قراءة تجميعٍ من عناوين على أنه نتيجة.
    basis_meaning_ar: str
    status: str = Field(pattern="^(" + "|".join(SYNTHESIS_STATUSES) + ")$")
    status_label_ar: str
    source_scope_summary: dict[str, int] = Field(default_factory=dict)
    supporting_count: int = 0
    contradicting_count: int = 0
    # **هل يبلغ كل سندٍ محتوًى خليةً بعينها؟** موضوعٌ لا يبلغ لا يُبنى عليه.
    is_traceable: bool = False
    generated_at: dt.datetime | None = None
    decided_at: dt.datetime | None = None


class ThemeTraceView(BaseModel):
    """المسار كاملًا: موضوع ← مراجع ← خلايا ← شواهد."""

    theme: ThemeView
    supporting: list[EvidenceLink] = Field(default_factory=list)
    contradicting: list[EvidenceLink] = Field(default_factory=list)
    note_ar: str = (
        "كل سطرٍ هنا يشير إلى خليةٍ في مصفوفة الأدبيات. افتح الخلية لترى نصّها "
        "وشاهدها ومدى ما قُرئ منه — والموضوع لا يقوم إلا على ما تراه."
    )


class ThemesView(BaseModel):
    project_id: uuid.UUID
    themes: list[ThemeView] = Field(default_factory=list)
    corpus_size: int = 0
    # **تحذيرٌ يُعرض حين يكون كلّه تجميعًا موضوعيًّا.**
    note_ar: str = ""
    basis_vocabulary: list[VocabularyEntry] = Field(default_factory=list)


# ═════════════════════ التعارضات ═════════════════════

class ContradictionSideView(BaseModel):
    """طرفٌ واحد — بسياقه كما سُجِّل، وبـ«غير مذكور» حيث لم يُسجَّل."""

    side: str = Field(pattern="^(a|b)$")
    source_id: uuid.UUID
    title: str
    result_ar: str
    direction: str = Field(pattern="^(" + "|".join(EFFECT_DIRECTIONS) + ")$")
    direction_label_ar: str
    significance: str = Field(pattern="^(" + "|".join(SIGNIFICANCE_STATES) + ")$")
    significance_label_ar: str
    population_ar: str | None = None
    country_ar: str | None = None
    method_ar: str | None = None
    measurement_ar: str | None = None
    period_year: int | None = None
    evidence_scope: str = Field(pattern="^(" + "|".join(SOURCE_SCOPES) + ")$")
    matrix_cell_id: uuid.UUID | None = None


class ContradictionView(BaseModel):
    """تعارضٌ محتمل — **بطرفيه كليهما وبسياق اختلافهما**، ولا يُسمّى أحدهما خطأ."""

    id: uuid.UUID
    construct_a_ar: str
    construct_b_ar: str | None = None
    relationship_ar: str
    conflict_kind: str = Field(pattern="^(" + "|".join(CONFLICT_KINDS) + ")$")
    conflict_label_ar: str
    context_divergence: list[str] = Field(default_factory=list)
    context_divergence_labels_ar: list[str] = Field(default_factory=list)
    context_explanation_ar: str | None = None
    status: str
    status_label_ar: str
    sides: list[ContradictionSideView] = Field(default_factory=list)
    generated_at: dt.datetime | None = None
    decided_at: dt.datetime | None = None


class ContradictionsView(BaseModel):
    project_id: uuid.UUID
    contradictions: list[ContradictionView] = Field(default_factory=list)
    corpus_size: int = 0
    note_ar: str = (
        "التعارض هنا اختلافُ نتيجتين على البناءات نفسها — لا اختلافُ صياغة ولا "
        "اختلافُ موضوع. ولا تُوصف أيّ دراسة بالخطأ: قد تكون النتيجتان صحيحتين "
        "كلٌّ في سياقها، ولذلك يُعرض السياق قبل الحكم."
    )


# ═════════════════════ الفجوات ═════════════════════

class SearchScopeView(BaseModel):
    """**مدى ما بُحث — يخرج مع كل فجوة، لا في صفحةٍ منفصلة.**"""

    indexes_searched: list[str] = Field(default_factory=list)
    # تبقى `False`: المنصّة لا تُجري بحثًا منهجيًّا في الفهارس هنا، والادّعاء
    # بغير ذلك يجعل «لم تظهر دراسة» تُقرأ نتيجةَ مسحٍ شامل.
    search_was_systematic: bool = False
    corpus_size: int = 0
    content_read: int = 0
    full_text_read: int = 0
    saved_not_screened: int = 0
    excluded: int = 0
    taken_at: str | None = None


class GapView(BaseModel):
    """فجوةٌ **محتملة** — ولا حقل هنا يقول إنها مؤكَّدة."""

    id: uuid.UUID
    gap_type: str = Field(pattern="^(" + "|".join(GAP_TYPES) + ")$")
    gap_type_label_ar: str
    description_ar: str
    why_suggested_ar: str
    known_limitations_ar: str
    strength: str = Field(pattern="^(" + "|".join(GAP_STRENGTHS) + ")$")
    strength_label_ar: str
    strength_meaning_ar: str
    sources_considered: int
    search_scope: SearchScopeView
    source_scope_distribution: dict[str, int] = Field(default_factory=dict)
    supporting: list[EvidenceLink] = Field(default_factory=list)
    contradicting: list[EvidenceLink] = Field(default_factory=list)
    considered: list[EvidenceLink] = Field(default_factory=list)
    contradiction_id: uuid.UUID | None = None
    status: str
    status_label_ar: str
    generated_at: dt.datetime | None = None
    decided_at: dt.datetime | None = None
    # فرصةٌ لا تُنشأ إلا من معتمَدة — والواجهة تعرف ذلك من الخادم لا بتخمين.
    may_become_opportunity: bool = False


class NotAssessedView(BaseModel):
    """**ما تعذّر الحكم فيه — يُعرض بقدر ما يُعرض ما حُكم فيه.**

    و`verdict` مفردةُ محرّك القواعد نفسها: `insufficient_information`. وقائمةُ
    فجواتٍ تذكر ما وجدته وتصمت عمّا عجزت عنه يقرأها الباحث «لا شيء آخر».
    """

    gap_type: str
    gap_type_label_ar: str
    verdict: str = Field(pattern="^insufficient_information$")
    reason_ar: str


class GapsView(BaseModel):
    project_id: uuid.UUID
    gaps: list[GapView] = Field(default_factory=list)
    not_assessed: list[NotAssessedView] = Field(default_factory=list)
    corpus_size: int = 0
    search_scope: SearchScopeView | None = None
    strength_vocabulary: list[VocabularyEntry] = Field(default_factory=list)
    note_ar: str = (
        "هذه فجواتٌ **محتملة** ضمن مجموعة مراجعك الحالية وحدها. لم يُجرِ النظام "
        "بحثًا منهجيًّا في الفهارس، فما لم يظهر هنا قد يكون موجودًا خارج مجموعتك. "
        "ولا تُكتب أيّ ملاحظةٍ منها في ورقةٍ بصيغة «لا توجد دراسات»."
    )


# ═════════════════════ الفرص البحثية ═════════════════════

class RelatedStudyView(BaseModel):
    source_id: uuid.UUID
    title: str
    role: str
    evidence_scope: str


class OpportunityPreviewView(BaseModel):
    """معاينةٌ **لا تكتب شيئًا** — سبعةُ أسئلةٍ بأجوبتها قبل أن يقرّر الباحث."""

    gap_candidate_id: uuid.UUID
    gap_type: str
    gap_type_label_ar: str
    what_we_noticed_ar: str
    why_it_might_matter_ar: str
    evidence_basis_ar: str
    related_studies: list[RelatedStudyView] = Field(default_factory=list)
    still_uncertain_ar: str
    strength_label_ar: str
    strength_meaning_ar: str
    next_step_ar: str
    editable_fields: list[str] = Field(default_factory=list)
    requires_confirmation: bool = True


class OpportunityCreateRequest(BaseModel):
    """إنشاءُ بطاقة — **بتأكيدٍ صريح، ومن فجوةٍ معتمَدة**.

    و`uncertainties_ar` إلزاميّ: بطاقةٌ بلا عدم يقينٍ معلن تُقرأ خطةً مثبتة،
    والقيد في القاعدة يرفضها أيضًا.
    """

    gap_candidate_id: uuid.UUID
    confirmed: bool = Field(description="تأكيدٌ صريح — ولا تُنشأ بطاقة بدونه")
    phenomenon_ar: str = Field(min_length=1, max_length=2000)
    context_ar: str | None = Field(default=None, max_length=2000)
    population_ar: str | None = Field(default=None, max_length=2000)
    constructs_ar: str | None = Field(default=None, max_length=2000)
    possible_contribution_ar: str = Field(min_length=1, max_length=4000)
    methodological_opportunity_ar: str | None = Field(default=None, max_length=4000)
    evidence_basis_ar: str = Field(min_length=1, max_length=4000)
    uncertainties_ar: str = Field(min_length=1, max_length=4000)


class OpportunityView(BaseModel):
    id: uuid.UUID
    gap_candidate_id: uuid.UUID
    gap_type: str
    gap_type_label_ar: str
    phenomenon_ar: str
    context_ar: str | None = None
    population_ar: str | None = None
    constructs_ar: str | None = None
    possible_contribution_ar: str
    methodological_opportunity_ar: str | None = None
    evidence_basis_ar: str
    uncertainties_ar: str
    created_at: dt.datetime | None = None
    spawned_project_id: uuid.UUID | None = None


class OpportunitiesView(BaseModel):
    project_id: uuid.UUID
    opportunities: list[OpportunityView] = Field(default_factory=list)
    note_ar: str = (
        "كل بطاقةٍ هنا كتبها باحثٌ بيده بعد أن اعتمد فجوةً محتملة. ولا تُولَّد "
        "بطاقةٌ تلقائيًّا، ولا تعني البطاقةُ أن الفجوة ثبتت."
    )


class ProjectPreviewView(BaseModel):
    """**ما سيقع بالضبط قبل أن يقع** — وما لن يقع، وما لن يتغيّر."""

    working_title_ar: str
    from_opportunity_id: uuid.UUID
    gap_type_label_ar: str
    will_create_ar: list[str] = Field(default_factory=list)
    will_not_create_ar: list[str] = Field(default_factory=list)
    unchanged_ar: list[str] = Field(default_factory=list)
    requires_confirmation: bool = True


class ProjectFromOpportunityRequest(BaseModel):
    """«إنشاء مشروع بحثي» — **معاينةٌ ثم تأكيد، ولا يقع بضغطةٍ واحدة**."""

    confirmed: bool
    working_title_ar: str = Field(min_length=1, max_length=500)


__all__ = [
    "ContradictionSideView",
    "ContradictionView",
    "ContradictionsView",
    "DecisionRequest",
    "EvidenceLink",
    "GapView",
    "GapsView",
    "NotAssessedView",
    "OpportunitiesView",
    "OpportunityCreateRequest",
    "OpportunityPreviewView",
    "OpportunityView",
    "ProjectFromOpportunityRequest",
    "ProjectPreviewView",
    "RelatedStudyView",
    "SearchScopeView",
    "ThemeTraceView",
    "ThemeView",
    "ThemesView",
    "VocabularyEntry",
]

File: apps/web/src/lib/synthesis.ts
/**
 * طبقة التركيب | Synthesis: themes, contradictions, gaps, opportunities.
 *
 * **الأسماء تأتي من الخادم مترجَمة.** فلا تخترع الواجهة اسمًا لمفردة، ولا
 * تبني جدول ترجمةٍ ثانيًا يفترق عن جدول الخادم بأول إضافة — فيظهر في
 * الشاشة `weak_signal` بدل «إشارة ضعيفة» ولا يفشل شيء.
 *
 * **والحدُّ يُعرض مع الدعوى دائمًا.** `sources_considered` و`search_scope`
 * و`known_limitations_ar` حقولٌ إلزامية في العقد، ولا تُعرض فجوةٌ بدونها:
 * وصفٌ بلا حدّه يُقرأ جملةً مطلقة، ثم يُكتب في ورقة.
 */
import { apiFetch } from "./api";
import type { Locale } from "./i18n";

/** دورة الحياة — مفردات المنصّة نفسها، ولا `UNSURE` ولا `MAYBE`. */
export type SynthesisStatus =
  | "generated"
  | "needs_review"
  | "approved"
  | "rejected"
  | "unknown";

/** ما يقبله الخادم حكمًا — و«مقترَح» ليست منها: قبولها محوُ مراجعة. */
export type Decision = Exclude<SynthesisStatus, "generated">;

/** **الفرق الذي لا يُطوى**: تجميعٌ من عناوين مقابل تركيبٍ من محتوًى مقروء. */
export type ThemeBasis = "topic_cluster" | "content_synthesis";

export type GapStrength = "weak_signal" | "emerging_pattern" | "supported_candidate";

export type SourceScope = "metadata_only" | "abstract_only" | "full_text";

export interface VocabularyEntry {
  key: string;
  label_ar: string;
  label_en: string;
  meaning_ar: string | null;
}

/** حلقةٌ في سلسلة الأثر: مرجع ← خلية ← شاهد. */
export interface EvidenceLink {
  source_id: string;
  title: string;
  role: "supporting" | "contradicting" | "considered";
  basis_field_key: string | null;
  /** غيابه يعني سندًا من بياناتٍ وصفية — وتلك حالٌ تُعلَن لا تُخفى. */
  matrix_cell_id: string | null;
  evidence_scope: SourceScope;
  evidence_quote: string | null;
  evidence_locator: string | null;
  cell_state: string | null;
  cell_value_ar: string | null;
}

export interface Theme {
  id: string;
  label_ar: string;
  description_ar: string | null;
  basis: ThemeBasis;
  basis_label_ar: string;
  basis_meaning_ar: string;
  status: SynthesisStatus;
  status_label_ar: string;
  source_scope_summary: Record<string, number>;
  supporting_count: number;
  contradicting_count: number;
  is_traceable: boolean;
  generated_at: string | null;
  decided_at: string | null;
}

export interface ThemesView {
  project_id: string;
  themes: Theme[];
  corpus_size: number;
  note_ar: string;
  basis_vocabulary: VocabularyEntry[];
}

export interface ThemeTrace {
  theme: Theme;
  supporting: EvidenceLink[];
  contradicting: EvidenceLink[];
  note_ar: string;
}

export interface ContradictionSide {
  side: "a" | "b";
  source_id: string;
  title: string;
  result_ar: string;
  direction: string;
  direction_label_ar: string;
  significance: string;
  significance_label_ar: string;
  population_ar: string | null;
  country_ar: string | null;
  method_ar: string | null;
  measurement_ar: string | null;
  period_year: number | null;
  evidence_scope: SourceScope;
  matrix_cell_id: string | null;
}

export interface Contradiction {
  id: string;
  construct_a_ar: string;
  construct_b_ar: string | null;
  relationship_ar: string;
  conflict_kind: string;
  conflict_label_ar: string;
  context_divergence: string[];
  context_divergence_labels_ar: string[];
  context_explanation_ar: string | null;
  status: SynthesisStatus;
  status_label_ar: string;
  sides: ContradictionSide[];
  generated_at: string | null;
  decided_at: string | null;
}

export interface ContradictionsView {
  project_id: string;
  contradictions: Contradiction[];
  corpus_size: number;
  note_ar: string;
}

export interface SearchScope {
  indexes_searched: string[];
  /** تبقى `false`: لا بحث منهجيّ في الفهارس، والادّعاء بغيره يقلب المعنى. */
  search_was_systematic: boolean;
  corpus_size: number;
  content_read: number;
  full_text_read: number;
  saved_not_screened: number;
  excluded: number;
  taken_at: string | null;
}

export interface Gap {
  id: string;
  gap_type: string;
  gap_type_label_ar: string;
  description_ar: string;
  why_suggested_ar: string;
  known_limitations_ar: string;
  strength: GapStrength;
  strength_label_ar: string;
  strength_meaning_ar: string;
  sources_considered: number;
  search_scope: SearchScope;
  source_scope_distribution: Record<string, number>;
  supporting: EvidenceLink[];
  contradicting: EvidenceLink[];
  considered: EvidenceLink[];
  contradiction_id: string | null;
  status: SynthesisStatus;
  status_label_ar: string;
  generated_at: string | null;
  decided_at: string | null;
  may_become_opportunity: boolean;
}

/** **ما تعذّر الحكم فيه** — يُعرض بقدر ما يُعرض ما وُجد. */
export interface NotAssessed {
  gap_type: string;
  gap_type_label_ar: string;
  verdict: "insufficient_information";
  reason_ar: string;
}

export interface GapsView {
  project_id: string;
  gaps: Gap[];
  not_assessed: NotAssessed[];
  corpus_size: number;
  search_scope: SearchScope | null;
  strength_vocabulary: VocabularyEntry[];
  note_ar: string;
}

export interface RelatedStudy {
  source_id: string;
  title: string;
  role: string;
  evidence_scope: SourceScope;
}

export interface OpportunityPreview {
  gap_candidate_id: string;
  gap_type: string;
  gap_type_label_ar: string;
  what_we_noticed_ar: string;
  why_it_might_matter_ar: string;
  evidence_basis_ar: string;
  related_studies: RelatedStudy[];
  still_uncertain_ar: string;
  strength_label_ar: string;
  strength_meaning_ar: string;
  next_step_ar: string;
  editable_fields: string[];
  requires_confirmation: boolean;
}

export interface Opportunity {
  id: string;
  gap_candidate_id: string;
  gap_type: string;
  gap_type_label_ar: string;
  phenomenon_ar: string;
  context_ar: string | null;
  population_ar: string | null;
  constructs_ar: string | null;
  possible_contribution_ar: string;
  methodological_opportunity_ar: string | null;
  evidence_basis_ar: string;
  uncertainties_ar: string;
  created_at: string | null;
  spawned_project_id: string | null;
}

export interface OpportunitiesView {
  project_id: string;
  opportunities: Opportunity[];
  note_ar: string;
}

export interface ProjectPreview {
  working_title_ar: string;
  from_opportunity_id: string;
  gap_type_label_ar: string;
  will_create_ar: string[];
  /** **ما لن يقع** — وبدونه يظنّ الباحث أن مراجعه ستُنقل. */
  will_not_create_ar: string[];
  unchanged_ar: string[];
  requires_confirmation: boolean;
}

const base = "/api/v1/synthesis";

export const loadThemes = (locale: Locale, projectId: string) =>
  apiFetch<ThemesView>(`${base}/projects/${projectId}/themes`, { locale });

export const loadThemeTrace = (locale: Locale, projectId: string, themeId: string) =>
  apiFetch<ThemeTrace>(`${base}/projects/${projectId}/themes/${themeId}/trace`, {
    locale,
  });

export const loadContradictions = (locale: Locale, projectId: string) =>
  apiFetch<ContradictionsView>(`${base}/projects/${projectId}/contradictions`, {
    locale,
  });

export const loadGaps = (locale: Locale, projectId: string) =>
  apiFetch<GapsView>(`${base}/projects/${projectId}/gaps`, { locale });

export const loadOpportunities = (locale: Locale, projectId: string) =>
  apiFetch<OpportunitiesView>(`${base}/projects/${projectId}/opportunities`, {
    locale,
  });

/**
 * إعادة التحليل — **ولا تمحو حكمًا قاله الباحث**.
 *
 * والخادم هو من يحرس ذلك (يستبدل ما لم يُحكم فيه وحده)، والواجهة تقول ذلك
 * للباحث قبل الضغط: زرٌّ لا يُعرف أثره يُضغط ثم يُندَم عليه.
 */
export const analyze = (locale: Locale, projectId: string) =>
  apiFetch<ThemesView>(`${base}/projects/${projectId}/analyze`, {
    locale,
    method: "POST",
  });

export const decideTheme = (
  locale: Locale,
  projectId: string,
  themeId: string,
  status: Decision,
) =>
  apiFetch<Theme>(`${base}/projects/${projectId}/themes/${themeId}/decision`, {
    locale,
    method: "POST",
    body: JSON.stringify({ status }),
  });

export const decideContradiction = (
  locale: Locale,
  projectId: string,
  contradictionId: string,
  status: Decision,
) =>
  apiFetch<{ status: Decision }>(
    `${base}/projects/${projectId}/contradictions/${contradictionId}/decision`,
    { locale, method: "POST", body: JSON.stringify({ status }) },
  );

export const decideGap = (
  locale: Locale,
  projectId: string,
  gapId: string,
  status: Decision,
) =>
  apiFetch<Gap>(`${base}/projects/${projectId}/gaps/${gapId}/decision`, {
    locale,
    method: "POST",
    body: JSON.stringify({ status }),
  });

export const loadOpportunityPreview = (
  locale: Locale,
  projectId: string,
  gapId: string,
) =>
  apiFetch<OpportunityPreview>(
    `${base}/projects/${projectId}/gaps/${gapId}/opportunity-preview`,
    { locale },
  );

/** **البطاقة لا تُنشأ بلا تأكيدٍ صريح** — والخادم يرفضها بدونه أيضًا. */
export const createOpportunity = (
  locale: Locale,
  projectId: string,
  card: {
    gap_candidate_id: string;
    confirmed: true;
    phenomenon_ar: string;
    context_ar?: string | null;
    population_ar?: string | null;
    constructs_ar?: string | null;
    possible_contribution_ar: string;
    methodological_opportunity_ar?: string | null;
    evidence_basis_ar: string;
    uncertainties_ar: string;
  },
) =>
  apiFetch<Opportunity>(`${base}/projects/${projectId}/opportunities`, {
    locale,
    method: "POST",
    body: JSON.stringify(card),
  });

export const loadProjectPreview = (
  locale: Locale,
  projectId: string,
  opportunityId: string,
) =>
  apiFetch<ProjectPreview>(
    `${base}/projects/${projectId}/opportunities/${opportunityId}/project-preview`,
    { locale },
  );

export const createProjectFromOpportunity = (
  locale: Locale,
  projectId: string,
  opportunityId: string,
  workingTitleAr: string,
) =>
  apiFetch<Opportunity>(
    `${base}/projects/${projectId}/opportunities/${opportunityId}/project`,
    {
      locale,
      method: "POST",
      body: JSON.stringify({ confirmed: true, working_title_ar: workingTitleAr }),
    },
  );

/**
 * وصفُ المرشَّح في سطر — **اسمًا مُعلَنًا لكل زرٍّ متكرّر**.
 *
 * وفي كل شاشةٍ هنا صفٌّ من أزرارٍ متطابقة الاسم: «اعتماد» بجانب «اعتماد».
 * فمن يتنقّل بلوحة المفاتيح أو يسمع الشاشة لا يميّز بينها إطلاقًا.
 */
export function describe(item: { label?: string; title?: string }): string {
  return item.label ?? item.title ?? "";
}
